webpack/scenes/ContentViews/Details/Repositories/ContentViewRepositories.js

```diff
@@ -60,6 +76,56 @@ const ContentViewRepositories = ({ cvId }) => {
   ];
   const loading = status === STATUS.PENDING;
 
+  const toggleBulkAction = () => {
+    setBulkActionOpen(!bulkActionOpen);
+  };
+  const onAdd = (repos) => {
+    const { repository_ids: repositoryIds } = details;
+    dispatch(updateContentView(cvId, { repository_ids: repositoryIds.concat(repos) }));
+  };
+
+  const onRemove = (repos) => {
+    const reposToDelete = [].concat(repos);
+    const { repository_ids: repositoryIds } = details;
+    const deletedRepos = repositoryIds.filter(x => !reposToDelete.includes(x));
+    dispatch(updateContentView(cvId, { repository_ids: deletedRepos }));
+  };
+
+  const addBulk = () => {
+    const reposToAdd = [];
+    rows.forEach(row => row.selected && reposToAdd.push(row.repoId));
+    setBulkActionOpen(false);
+    onAdd(reposToAdd);
+  };
+
+  const removeBulk = () => {
+    const reposToDelete = [];
+    rows.forEach(row => row.selected && reposToDelete.push(row.repoId));
+    setBulkActionOpen(false);
+    onRemove(reposToDelete);
+  };
+
+  const actionResolver = (rowData, { _rowIndex }) => {
+    if (rowData.parent || rowData.compoundParent || rowData.noactions) return null;
+    const { repository_ids: repositoryIds } = details;
+    return [
+      {
+        title: 'Add',
+        isDisabled: repositoryIds && repositoryIds.includes(rowData.repoId),
+        onClick: (_event, rowId, rowInfo) => {
+          onAdd(rowInfo.repoId);
+        },
+      },
+      {
+        title: 'Remove',
+        isDisabled: repositoryIds && !repositoryIds.includes(rowData.repoId),
+        onClick: (_event, rowId, rowInfo) => {
+          onRemove(rowInfo.repoId);
+        },
+      },
+    ];
+  };
+
```

Comment:
can you move all of these functions underneath the `useEffect` hooks? see [react docs]( https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level). They can move unless they are used in the effect hooks.

(`getCVReposWithOptions` can move as well, that was my mistake!)